sklearn/decomposition/pca.py

```diff
@@ -383,6 +383,13 @@ def _fit(self, X):
         else:
             n_components = self.n_components
 
+        if n_components != "mle" and \
+                (n_components > 1 and
+                 not (np.issubdtype(type(n_components), np.integer))):
+            raise ValueError("n_components=%r must be of type int "
+                             "when bigger than 1, was of type=%r"
```

Comment:
I think it might worth to mention greater than or equal to 1 in this case.

Reply:
Thought it would fall on the the float case, but it has to be lesser than 1, but just checked you are right. My bad. Will fix.